### PR TITLE
Switch to ES6

### PR DESCRIPTION
**Pull request by blx**
- enables the `es2015` babel preset (everything was being babelified for jsx already anyway)
- adds jasmine (`gulp jasmine`) and a whopping four test cases
- should and seems to behave 100% the same way (no api changes)
- adds missing `key` and `htmlFor` attrs in Search component
- minor code translations: const/let, array spread, arrow fns (removed most `.bind(this)`), imports, format strings

🐲

### CODE REVIEW

File: ubyssey/static/src/js/components/Search.jsx [hunk 1]
```diff
@@ -37,36 +38,33 @@ var Search = React.createClass({
...
         return (
             <div className="u-container">
                 <form method="get" action="/archive/">
-                    <label for="author-search"><i className="fa fa-search"></i></label>
-                    <input ref="search" className={this.state.results.length > 0 ? "open" : ""} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
+                    <label htmlFor="author-search"><i className="fa fa-search"></i></label>
+                    <input ref="search" className={this.state.results.length > 0 && "open"} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
                 </form>
-                <ul className={"results" + (this.state.results.length > 0 ? " open" : "")}>{results}</ul>
+                <ul className={"results" + (this.state.results.length > 0 && " open")}>{results}</ul>
```
Comment (psiemens) [inline ubyssey/static/src/js/components/Search.jsx]: `(this.state.results.length > 0 && " open")`

This appends "false" to the className when `results.length == 0`.

File: ubyssey/static/src/js/components/Search.jsx [hunk 2]
```diff
@@ -37,36 +38,33 @@ var Search = React.createClass({
...
+            <li key={i}><a href={item.url}>{item.headline}</a></li>
+        ));
 
         return (
             <div className="u-container">
                 <form method="get" action="/archive/">
-                    <label for="author-search"><i className="fa fa-search"></i></label>
-                    <input ref="search" className={this.state.results.length > 0 ? "open" : ""} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
+                    <label htmlFor="author-search"><i className="fa fa-search"></i></label>
+                    <input ref="search" className={this.state.results.length > 0 && "open"} name="q" id="search-bar" type="text" autoComplete="off" onChange={this.updateQuery} value={this.state.q} placeholder="Search The Ubyssey..." />
```
Comment (psiemens) [inline ubyssey/static/src/js/components/Search.jsx]: Same as below 👇

File: ubyssey/static/gulpfile.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
+require('babel-register'); // Pass require()s through babel (for running jasmine)
```
Comment (psiemens) [inline ubyssey/static/gulpfile.js:1]: `babel-register` is missing in `package.json`

**Review by psiemens — CHANGES_REQUESTED** [review]

File: ubyssey/static/src/js/components/Search.jsx (hunk 1 above)
Comment (blx) [reply ubyssey/static/src/js/components/Search.jsx]: yeah my bad, this works for conditionally rendering JSX but not strings

File: ubyssey/static/gulpfile.js (hunk 3 above)
Comment (blx) [reply ubyssey/static/gulpfile.js:1]: good catch, updated.

**Review by psiemens — APPROVED** [review]

**Comment by psiemens** [comment]
Everything *seems* to be working (I'm pretty sure I tested every component on the site).

Merging into staging for now -- will test again before next release.

**Comment by blx** [comment]
Cool! I tried to test things as I went too, and the translations were largely mechanical, so there shouldn't be too many dragons lurking. 🤔

**Pull request merged**